src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    switch (action['type']) {
+      case 'addProperties':
+        Object.assign(state, action['extraData']);
+        break;
+
+      case 'removeProperties':
+        action['keysToRemove'].forEach(key => delete state[key]);
```

Comment:
You can use object._keyname_ but not object['_keyname_'] when you know exactly which key do you have. It would work the same way but it is a better practice if you know exactly name of key and it doesn't start with nonletter character.
Good job)